runtime/lua/vim/filetype/detect.lua

```diff
@@ -787,9 +862,67 @@ function M.scd(bufnr)
   end
 end
 
-function M.sh(path, bufnr) end
+-- Also called from filetype.lua
+function M.sh(path, bufnr, name)
```

Comment:
In contradiction to the general rule, I think we can use case *sensitive* regexes (i.e. remove `\c`) in this function, because:

1) a lot of times the `name` argument is being passed directly from `filetype.lua`, where we know the value is lower case
2) the rest of the cases come from a shebang line, which we know will be lower case (I guess *technically* on a case insensitive filesystem like macOS someone could use an uppercase shebang, but I don't care about supporting that kind of madness)